packages/xod-client-electron/src/debugger/tetheringInetMiddleware.js

```diff
@@ -0,0 +1,152 @@
+import * as R from 'ramda';
+import AtNet from 'at-internet';
+import client from 'xod-client';
+import { allPromises } from 'xod-func-tools';
+import { ipcRenderer } from 'electron';
+
+import { DEBUG_SERIAL_SEND } from '../shared/events';
+
+const DELAY_BETWEEN_CHUNKS = 500;
+
+const formatTetherinErrorMessage = err => ({
+  title: 'Tethering Internet Error',
+  note: err.message,
+});
+
+// Mutable list of chunks to send
+// New chunks are placed in the end of list
+//
+// Transmit function (created by `createTransmitter`) is shifting
+// chunks and transmit it using `sender` function, which are different
+// for Simulation and Debug mode.
+//
+let chunksToSend = [];
+
+// Places the new chunks in the end of mutable `chunksToSend` list (SIDE-EFFECT)
+// :: [String] -> [String]
+const queueChunks = newChunks => {
+  chunksToSend = R.concat(chunksToSend, newChunks);
+  return chunksToSend;
+};
+
+// :: (String -> _) -> (_ -> _)
+const createTransmitter = sender => {
+  let inProcess = false;
+
+  const transmit = (force = false) => {
+    if (chunksToSend.length === 0) {
+      inProcess = false;
+      return;
+    }
+    if (inProcess && !force) return;
+
+    inProcess = true;
+    const nextChunk = chunksToSend.shift();
+    setTimeout(() => {
+      // console.log('<', nextChunk);
+      sender(nextChunk);
+      transmit(chunksToSend, true);
+    }, DELAY_BETWEEN_CHUNKS);
+  };
+
+  return transmit;
+};
+
+// :: String -> [String]
+// Pack data into chunks, if needed:
+// - Simple AT answers just placed inside array
+// - IPD (Incoming Package Data) is splited on chunks of max 64 bytes length
+//   with prepended prefix `+XOD:{nodeId}:{dataLength}:`
```

Comment:
No, it should not.
Other tweaks have different nodeId.
A tethering internet node always waits for the chunks that begin with this header. After parsing this header it will read up any characters until it read the specified in header amount of bytes (even the chunk is: `+XOD:42:14:+XOD:42:4:lala` it will take the second part as a content of the chunk).